### ISSUE TITLE
Feature request: set `sys.stdout = None` with `--disable-console`

### ISSUE THREAD

**Issue by bersbersbers** [Nuitka/Nuitka#2083]
On Linux, one can usually print stuff to a console. In Windows, with `--disable-console`, one cannot - at least, no one will see it.

With `pyinstaller`, `sys.stdout is None` in this case, which can be used to show output by other means (write to a log file, show a message box, ...). This seems to be in line with what `pythonw.exe` does, according to their documentation: https://github.com/pyinstaller/pyinstaller/commit/89c120e0ac3694a8707c5e850c0246328f632ac1

Also compare:
```
python.exe  -c "import sys; f = open('stdout_python.txt',  'w'); f.write(str(sys.stdout))" 
pythonw.exe -c "import sys; f = open('stdout_pythonw.txt', 'w'); f.write(str(sys.stdout))" 

more stdout_*.txt
<_io.TextIOWrapper name='<stdout>' mode='w' encoding='utf-8'>
None
```
So it seems to quite standard to have with `sys.stdout == None` in some cases.

Long story short, I think it would be useful to have `sys.stdout = None` with `--disable-console` in Nuitka as well.

(Same for `stderr`, I guess.)

**Comment by kayhayen** [comment]
The fact that `pythonw.exe` does this is indeed evidence of what it should be. However, I am not convinced that the reason people disable the console is that they tested their program with `pythonw.exe` or even know what it is.

But I accept that definitely this is something that should be supported. Maybe the forced output paths could allow doing that, since that will also affect non-Windows.

I think, it's also a shortcoming of `__compiled__` that these modes are not all recognizable there. Otherwise a workaound that distiguishes console vs. no-console mode would be doable.

**Comment by kayhayen** [comment]
With #2038 we will probably revamp how console is handled, would be nice to get the options naming sane at that point.

**Comment by bersbersbers** [comment]
> I think, it's also a shortcoming of __compiled__ that these modes are not all recognizable there. Otherwise a workaound that distiguishes console vs. no-console mode would be doable.

I generally agree with you that code should not need to know whether it runs in Nuitka or not (have read this elsewhere), so I don't actually think it *has* to be recognizable. I usually recognize if `sys.stdout` is useable by, well, checking `sys.stdout is None`. If that is ensured, I am happy :) Everything else is icing on the cake, but for the record, nothing that I would need. (Writing portable code implies one should only do things that one could do, for example, in `pythonw.exe` as well.)

**Comment by kayhayen** [comment]
Well, Nuitka has made `sys.stdout` usable. It redirects to `NUL:` though, which is barely a problem.

Compatibility is paramount to Nuitka, otherwise the differences get arbitrarily hard to debug.

**Comment by bersbersbers** [comment]
> Well, Nuitka has made sys.stdout usable.

Yes, sure - I should have been more exact. I probably meant "useful" :) while `sys.stdout` redirecting to `NUL` is usable, it's hardly useful, especially if it's not easy to see from the outside *if* is redirecting. Anyway, using `__compile__` works of course - not super-elegant, but pragmatic enough.

**Comment by kayhayen** [comment]
Unless of course, you as a Python developer test your program with `python.exe` and one day, maybe after a dependency upgrade, the compile program starts crashing. I am defending it as a default, not as the only offer naturally. I guess this and the solution will be part of `typical problems` in the user manual soon.

But no worries, this option will soon be available. I just need to make my mind up, how the whole console business should be arranged. I am thinking that `None` may just be a value for the forced outputs in some form, obviously specific to these options I guess. 

For obvious reasons, Nuitka doesn't really want to run with `pythonw.exe` itself (even raises an error dialog box if you try), so there has be a mode where you indicate that you want that.

**Comment by kayhayen** [comment]
I added support for the following things to factory version.

For all platforms, now `--force-stdout-spec="%NONE%"` will put the None value to `sys.stdout`. There is also `--force-stderr-spec` for `sys.stderr`.

Also, for all platforms, now `%NULL%` is available as a value too. That puts a file backed by `os.devnull` there. Previously and still now, this is done when disabling the console on Windows.

The forcing options do not yet cover `sys.stdin`. That is still doing the `os.devnull` stuff. I might add this as a feature too in the future, but I wasn't immediately aware if this even makes sense for `None`. I added that to avoid crashes in the past as well.

**Comment by kayhayen** [comment]
Thanks for your report, this is fixed on the factory branch, which is a development version under rapid development. You can try it out by going here: https://nuitka.net/doc/factory.html 

Feedback if this is working is very welcome, just please do not share plans of doing it, but rather confirmations or denials of it working.

**Comment by bersbersbers** [comment]
For the record: this is working fine on the develop branch. Thanks!

**Comment by bersbersbers** [comment]
Since I am looking forward to this feature, may I ask if 1.6 will be released soon (as indicated by the recent rc9) or is that still far out (as indicated by the milestone, https://github.com/Nuitka/Nuitka/issues?q=is%3Aopen+is%3Aissue+milestone%3A1.6)?

**Comment by kayhayen** [comment]
This is part of the stable release 1.6 that I just made.

**Comment by bersbersbers** [comment]
Beautiful, thanks ;)
